docs/deployment/overview.md

```diff
@@ -1,47 +1,254 @@
+| `Aspire.Hosting.Kubernetes` | Kubernetes | Yes | No | Apply with `kubectl`, GitOps, or other controllers. |
+| `Aspire.Hosting.Azure.AppContainers` | Azure Container Apps | Yes | Yes (Preview) | Deploy capability is in Preview and may change. |
+| `Aspire.Hosting.Azure.AppService` | Azure App Service | Yes | Yes (Preview) | Deploy capability is in Preview and may change. |
+
+**Compute environments:**
+
+- `AddDockerComposeEnvironment(...)` - Docker Compose deployment target
+- `AddKubernetesEnvironment(...)` - Kubernetes deployment target  
+- `AddAzureContainerAppEnvironment(...)` - Azure Container Apps deployment target
+- `AddAzureAppServiceEnvironment(...)` - Azure App Service deployment target
+
+> Deploy support is integration-specific. Absence of deploy support means you use the published artifacts with external tooling.
+
+## Typical workflows
+
+### 1. Generate artifacts (any integration)
+
+```bash
+aspire publish -o artifacts/
+```
+
+Review the contents of `artifacts/` (for example: Docker Compose files, Kubernetes manifests, Azure specification documents, etc.).
+
+### 2. Run locally (Docker example)
+
+```bash
+# Provide or export required environment variables, then:
+docker compose -f artifacts/docker-compose.yml up --build
+```
+
+Missing variables like `PG_PASSWORD` must be set in the shell, an `.env` file, or injected by your chosen runner.
+
+### 3. CI/CD pipeline pattern
+
+1. Run `aspire publish -o artifacts/` in a build job.
+2. Archive or inspect artifacts (linting, security scanning, image signing).
+3. Inject parameter values (environment variables, secrets store, variable groups).
+4. Execute deployment:
+   - Docker: `docker compose up -d`
+   - Kubernetes: `kubectl apply -f artifacts/`
+   - Azure: Use Azure CLI or deployment scripts with the published artifacts
+   - Custom: your own script or orchestrator.
+
+Alternative for Azure: If using integrations with deploy support, you can use `aspire deploy` instead of publishing and deploying separately.
+
+### 4. Using `aspire deploy`
+
+If an integration supports deployment, you can run:
+
+```bash
+aspire deploy
+```
+
+This resolves parameters and applies deployment changes for integrations that support deployment.
+
+## Extensibility
+
+The system is designed so you can author new integrations that participate in:
+
+- Publish phase: generating artifacts for a target platform (for example, a service mesh config, an infrastructure-as-code definition, or a proprietary orchestrator spec).
+- Deploy phase: applying resources to a new deployment target (for example, a proprietary app hosting environment).
+
+A custom integration can:
+
+- Transform application resources into platform-specific configurations.
+- Handle parameter resolution and secret management.
+- Implement deployment workflows for custom platforms.
+
+Because published assets are parameterized, you can integrate custom secret and configuration resolution without modifying the core publishing logic.
```

Comment:
Tihs section should be focused around adding `PublishingCallbackAnnotation` and `DeployingCallbackAnnotation` to resources which give control over defining what happens during aspire publish or aspire  deploy. This is still in preview and we expect changes to the design.

Use the example and text from https://github.com/dotnet/docs-aspire/blob/main/docs/whats-new/dotnet-aspire-9.4.md#aspire-deploy in this section.